sockeye/train.py

```diff
@@ -478,6 +492,8 @@ def create_decoder_config(args: argparse.Namespace, encoder_num_hidden: int,
     config_decoder = None  # type: Optional[Config]
 
     if args.decoder == C.TRANSFORMER_TYPE:
+        if args.decoder_only:
+            raise NotImplementedError()
```

Comment:
Nice work checking and raising the appropriate error!  We can also add checks here to notify users as soon as possible if they try to use incompatible settings: https://github.com/awslabs/sockeye/blob/master/sockeye/train.py#L75

Reply:
Done.